tests/by-util/test_expr.rs

```diff
@@ -116,6 +131,94 @@ fn test_parenthesis() {
         .stderr_only("expr: syntax error: unexpected argument '('\n");
 }
 
+#[test]
+fn test_missing_parenthesis() {
+    new_ucmd!()
+        .args(&["(", "2"])
+        .fails()
+        .code_is(2)
+        .stderr_only("expr: syntax error: expecting ')' after '2'\n");
+
+    new_ucmd!()
+        .args(&["(", "2", "a"])
+        .fails()
+        .code_is(2)
+        .stderr_only("expr: syntax error: expecting ')' instead of 'a'\n");
+}
+
+#[test]
+fn test_regexp_unmatched() {
+    new_ucmd!()
+        .args(&["_", ":", "a\\("])
+        .fails()
+        .code_is(2)
+        .stderr_only("expr: Unmatched ( or \\(\n");
+
+    new_ucmd!()
+        .args(&["_", ":", "a\\)"])
+        .fails()
+        .code_is(2)
+        .stderr_only("expr: Unmatched ) or \\)\n");
+
+    new_ucmd!()
+        .args(&["_", ":", "a\\{1"])
+        .fails()
+        .code_is(2)
+        .stderr_only("expr: Unmatched \\{\n");
+
+    new_ucmd!()
+        .args(&["_", ":", "a\\}1"])
+        .fails()
+        .code_is(2)
+        .stderr_only("expr: Unmatched \\}\n");
```

Comment:
I'm not sure about this test, with GNU `expr` I don't get an error. It simply outputs `0`:
```
$ expr _ : a\\\}1
0
$ echo $?
1
```